spec/requests/products_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe "Products", type: :request do
+  let(:user) { FactoryBot.create(:user, confirmed_at: Time.now) }  # Create a user and confirm their account
+
+  before do
+    sign_in user  # Sign in the user before each test
+  end
+
+  describe "GET #index" do
```

Comment:
Do we have other actions for products ? If yes then need to specs for those as well